public/app/features/alerting/unified/components/rules/deleted-rules/ConfirmRestoreDeletedRuleModal.tsx

```diff
@@ -0,0 +1,171 @@
+        onDismiss();
+        onRestoreSucess();
+      })
+      .catch((err) => {
+        onRestoreError(err);
+      });
+  }
+
+  async function onManualRestore() {
+    if (!ruleToRestore) {
+      return;
+    }
+    if (!rulerRuleType.grafana.rule(ruleToRestore)) {
+      return;
+    }
+    const namespaceName = await backendSrv
+      .getFolderByUid(ruleToRestore.grafana_alert.namespace_uid)
+      .then((folder) => folder.title);
+    createAlert(ruleToRestore, namespaceName);
+  }
+
+  return (
+    <ConfirmModal
+      isOpen={isOpen}
+      title={title}
+      confirmText={confirmText}
+      modalClass={styles.modal}
+      confirmButtonVariant={!error ? 'destructive' : 'primary'}
+      body={
+        <Stack direction="column" gap={2}>
+          <Trans i18nKey="alerting.deletedRules.restore-modal.body">
+            Are you sure you want to restore this deleted alert rule definition?
+          </Trans>
+
+          <div>{ruleToRestore && <RulePeview rule={ruleToRestore} />}</div>
+          {error && (
+            <Alert severity="warning" title={errorTitle}>
+              <Trans i18nKey="alerting.alertVersionHistory.restore-deleted-manually">
+                Your alert rule could not be restored. This may be due to changes to other entities such as contact
+                points, data sources etc. Please manually restore the deleted rule.
+              </Trans>
+              <pre style={{ marginBottom: 0 }}>
+                <code>{stringifyErrorLike(error)}</code>
+              </pre>
+            </Alert>
+          )}
+        </Stack>
+      }
+      onConfirm={!error ? onRestoreConfirm : onManualRestore}
+      onDismiss={onDismiss}
+    />
+  );
+};
+
+function RulePeview({ rule }: { rule: RulerRuleDTO }) {
+  const styles = useStyles2(getStyles);
+  return (
+    <div className={styles.content}>
+      <CodeEditor
+        width="100%"
+        height={600}
+        language={'json'}
+        value={JSON.stringify(rule, null, 4)}
+        monacoOptions={{
+          minimap: {
+            enabled: false,
+          },
+          scrollBeyondLastLine: false,
+          lineNumbers: 'on',
+          readOnly: true,
+        }}
+      />
+    </div>
+  );
+}
+
+const getStyles = () => ({
+  content: css({
+    flex: '1 1 100%',
+  }),
+  modal: css({
+    width: '700px',
+  }),
+});
+
+export function useRestoreDeletedRule() {
+  const [addRuleToRuleGroup] = useAddRuleToRuleGroup();
+
+  return useAsync(async (deletedRule: RulerGrafanaRuleDTO<GrafanaRuleDefinition>) => {
+    const ruleGroupIdentifier = {
+      dataSourceName: GRAFANA_RULES_SOURCE_NAME,
+      namespaceName: deletedRule.grafana_alert.namespace_uid,
+      groupName: deletedRule.grafana_alert.rule_group,
+    };
+    // save the new rule to the rule group
+    return addRuleToRuleGroup.execute(ruleGroupIdentifier, deletedRule);
+  });
+}
+
+const createAlert = async (ruleToRecover: RulerGrafanaRuleDTO, namespace: string) => {
```

Comment:
Should we rename to something like `redirectToRestoreForm`?
Maybe this function could fetch the namespace name from Uid, to keep most of the implementation in this function?